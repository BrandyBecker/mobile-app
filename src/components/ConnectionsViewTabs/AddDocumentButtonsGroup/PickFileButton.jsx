import React from 'react';
import Button from './Button.jsx';
import PickFileIcon from './PickFileIcon.jsx';
import * as DocumentPicker from 'expo-document-picker';
import convertFileToMedia from './convertFileToMedia';

/**********************************************************/

export default function PickFileButton({ afterAccept }) {

    function pickFile() {
        DocumentPicker
            .getDocumentAsync({
                /* MIME type */
                type: '*/*',
                /* whether to cache file for other expo APIs to use */
                copyToCacheDirectory: false,
                /* whether to let user select multiple files */
                multiple: false,
            })
            .then ((file) => {

                if (file.type === 'success') {
                    afterAccept(convertFileToMedia(file));
                }

            })
            .catch ((error) => {

                console.error('--- There was a problem selecting a file. ---');
                console.log(error);

            });
    }

    return (
        <Button onPress={pickFile}>
            <PickFileIcon/>
        </Button>
    );
}
