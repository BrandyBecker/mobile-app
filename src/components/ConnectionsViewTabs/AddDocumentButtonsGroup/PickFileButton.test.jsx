import React from 'react';
import PickFileButton from './PickFileButton';
import renderer from 'react-test-renderer';


it('renders correctly', () => {
    const tree = renderer
        .create(<PickFileButton/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
