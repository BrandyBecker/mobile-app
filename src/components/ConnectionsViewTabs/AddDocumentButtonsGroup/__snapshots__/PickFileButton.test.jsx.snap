// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#0279AC",
      "borderRadius": 16,
      "color": "#ffffff",
      "flexGrow": 0,
      "margin": 4,
      "opacity": 1,
      "padding": 8,
    }
  }
>
  <Text />
</View>
`;
